src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanManager.java

```diff
@@ -61,7 +60,7 @@ public Plan getPlan() {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(List<Block> dirtiedAssets) {
```

Comment:
I would expect that in the `DefaultPlanManager` the default behavior is to treat any dirtied Asset as indicating that it should not be worked on by another `PlanManager`.  So I would expect us to return `Optional.empty()` here if the Block we'd ordinarily like to return is in the `dirtiedAssets` list.